### PR TITLE
Remove jobs from registry.Service

### PR DESCRIPTION
**Pull request by crosbymichael**
This makes `registry.Service` a first class type and does not use jobs
to interact with this type.

Signed-off-by: Michael Crosby crosbymichael@gmail.com

### CODE REVIEW

**Comment by jfrazelle** [comment]
:dancer:

**Comment by jfrazelle** [comment]
omg its so pretty LGTM

**Comment by dmp42** [comment]
LGTM!

File: api/client/search.go [hunk 1]
```diff
@@ -49,6 +49,7 @@ func (cli *DockerCli) CmdSearch(args ...string) error {
 	if _, err := outs.ReadListFrom(rawBody); err != nil {
 		return err
 	}
+	outs.ReverseSort()
```
Comment (icecrime) [inline api/client/search.go]: Commit artifact?

**Comment by icecrime** [comment]
That's so much simpler <3 One nit, otherwise LGTM.

File: daemon/daemon.go [hunk 2]
```diff
@@ -793,15 +795,15 @@ func (daemon *Daemon) RegisterLinks(container *Container, hostConfig *runconfig.
...
+func NewDaemon(config *Config, eng *engine.Engine, registryService *registry.Service) (*Daemon, error) {
+	daemon, err := NewDaemonFromDirectory(config, eng, registryService)
 	if err != nil {
 		return nil, err
 	}
 	return daemon, nil
 }
 
-func NewDaemonFromDirectory(config *Config, eng *engine.Engine) (*Daemon, error) {
+func NewDaemonFromDirectory(config *Config, eng *engine.Engine, registryService *registry.Service) (*Daemon, error) {
```
Comment (tiborvass) [inline daemon/daemon.go]: @crosbymichael Also NewDaemonFromDirectory is not used, you can rename it to NewDaemon.

**Comment by tiborvass** [comment]
@crosbymichael thanks, much simpler with registry.Service in Daemon. I'll do that for the other things!

File: api/client/search.go (hunk 1 above)
Comment (crosbymichael) [inline api/client/search.go]: no

File: daemon/daemon.go (hunk 2 above)
Comment (crosbymichael) [inline daemon/daemon.go]: is it worth the CI time?

Comment (tiborvass) [inline daemon/daemon.go]: fair enough :)

**Comment by tiborvass** [comment]
LGTM

**Pull request merged**